rules/evasion/fake-process-name.yara

```diff
@@ -7,10 +6,14 @@ rule fake_kworker_val : critical {
     hash_2023_Unix_Downloader_Rocke_6107 = "61075056b46d001e2e08f7e5de3fb9bfa2aabf8fb948c41c62666fd4fab1040f"
   strings:
     $kworker = /\[{0,1}kworker\/[\d:\]]{1,5}/
-    $kworker2 = "kworker" fullword
+    $kworker2 = /\bkworker\b/
     $kworker3 = "[kworker"
+    // datadog process-agent
+    $ignore_ref = /(data->|[!\*\.])?is_kworker(\.\w{0,32})?/
```

Comment:
Can these expressions be simplified? Is there a reason why a single string (non-regex) isn't sufficient to exclude datadog?

Reply:
Good point -- I was overthinking this. I'll just ignore `DataDog` and `datadog`.

Reply:
Both `DataDog` and `datadog` were present so I used a small expression to cover both cases in [`d34ecb7` (#300)](https://github.com/chainguard-dev/bincapz/pull/300/commits/d34ecb7b8a48db5055013de6c39b39e59db6eab3).

Reply:
```
/bincapz # yara rules/evasion/fake-process-name.yara /usr/bin/process-agent -s
/bincapz # yara rules/evasion/fake-process-name.yara /usr/bin/process-agent -c
0
```